ReSharper.FSharp/test/data/parsing/Types - Type provider constructor 02 - named constant.fs.gold

```diff
@@ -0,0 +1,39 @@
+﻿Language: PsiLanguageType:F#
+IFSharpImplFile
+  IAnonModuleDeclaration
+    ITypeDeclarationGroup
+      FSharpTokenType+TypeTokenElement(type:TYPE, text:type)
+      Whitespace(type:WHITE_SPACE, text: ) spaces:" "
+      ITypeAbbreviationDeclaration
+        FSharpIdentifierToken(type:IDENTIFIER, text:T)
+        Whitespace(type:WHITE_SPACE, text: ) spaces:" "
+        FSharpTokenType+EqualsTokenElement(type:EQUALS, text:=)
+        Whitespace(type:WHITE_SPACE, text: ) spaces:" "
+        INamedType
+          ITypeReferenceName
+            FSharpIdentifierToken(type:IDENTIFIER, text:TypeProvider)
+            IPrefixAppTypeArgumentList
+              FSharpTokenType+LessTokenElement(type:LESS, text:<)
+              IStaticConstantNamed
+                INamedType
+                  ITypeReferenceName
+                    FSharpIdentifierToken(type:IDENTIFIER, text:arg1)
```

Comment:
Is it always a simple reference name? If yes, could we assert it in the tree builder and make a simple psi rule?